Jenkinsfile

```diff
@@ -11,14 +11,60 @@ pipeline {
 
     environment {
         // GLobal Vars
-        BACKEND_IMAGE_NAME = "dsd-italent-backend"
-        FRONTEND_IMAGE_NAME = "dsd-italent-frontend"
-        BACKEND_DIR = "services/backend"
-        FRONTEND_DIR = "services/frontend-v3"
-        NODE_ENV = "production"
+        BACKEND_IMAGE_NAME = 'dsd-italent-backend'
+        FRONTEND_IMAGE_NAME = 'dsd-italent-frontend'
+        BACKEND_DIR = 'services/backend'
+        FRONTEND_DIR = 'services/frontend-v3'
+        FRONTEND_DIR_I18 = 'services/frontend-v3/src/i18n'
+        NODE_ENV = 'production'
     }
 
     stages {
+        stage('run-tests'){
+            parallel{
+                stage('i18-check') {
+                    steps {
+                        dir("${FRONTEND_DIR_I18}") {
+                            sh 'npm init -y'
+                            sh 'npm i lodash'
+                            sh 'node check'
+                        }
+                    }
+                }        
+
+                stage('frontend') {
+                    steps {
+                        dir("${FRONTEND_DIR}") {
+                            sh """
+                                unset NPM_CONFIG_PREFIX
+                                source $NVM_DIR/nvm.sh
+                                nvm install "12.6.0"
+                                npm i yarn -g
```

Comment:
would it be possible to do this only once (like once for the backend and the frontend)?

Reply:
From what I understood from derek is that you need to set it up in every sh or else it would reset. We could merge both backend and frontend together but then we wont know what failed the build. You would have to go through the text to find out.